src/core/symbology/qgsmarkersymbollayer.cpp

```diff
@@ -3608,7 +3615,15 @@ QString QgsFontMarkerSymbolLayer::characterToRender( QgsSymbolRenderContext &con
     if ( stringToRender != mString )
     {
       charWidth = mFontMetrics->horizontalAdvance( stringToRender );
-      charOffset = QPointF( charWidth / 2.0, -mFontMetrics->ascent() / 2.0 );
+      if ( mVerticalAnchorPoint == VerticalAnchorPoint::Baseline )
+      {
+        const double sizePixels = context.renderContext().convertToPainterUnits( mSize, mSizeUnit, mSizeMapUnitScale );
+        charOffset = QPointF( charWidth / 2.0, -sizePixels / 2.0 );
+      }
+      else
+      {
+        charOffset = QPointF( charWidth / 2.0, -mFontMetrics->ascent() / 2.0 );
+      }
```

Comment:
I think you may also need to update QgsFontMarkerSymbolLayer::bounds

Reply:
QgsFontMarkerSymbolLayer::bounds uses this function as well, no? Here https://github.com/qgis/QGIS/blob/5baaf69614c0a46c318902981b47efbed34d4923/src/core/symbology/qgsmarkersymbollayer.cpp#L3928

Reply:
Yes, but "chrOffset" isn't used anywhere in the bounds function

Reply:
Yes. That's right, sorry. Anyway. I don't see where it should need this correction in this function. It's already using the scaledSize here (and not the mFontMetrics, which was wrong in the other functions). Or do I don't see the obvious?

The only thing we could do is making it behave consequently wrong with the bottom setting. But this would break backwards compatibility with the projects, and that is the only reason we're keeping the original way (with bottom) and not replacing it with this bug fix at all until it can be replaced by the bounds-implementation you suggested.

Reply:
Can you adapt the  TestQgsFontMarkerSymbol::bounds() test to check other vertical alignments?

Reply:
Okay, I did that. When we compare an object in the result image it looks like this.

![bounds-compare](https://github.com/user-attachments/assets/64f615b6-7fbc-4f57-a343-2558f4e4173a)

Honestly, I'm a little bit confused and I don't know if I understand everything right. I expected the bounds to be the same on baseline and bottom. Since the offset of the anchor is calculated in `calculateOffsetAndRotation` (as always) and the positioning - that is fixed in the other functions on baseline with this implementation - is here in the bounds always `-scaledSize / 2.0` (no matter if baseline or not).

Reply:
@nyalldawson Is this okay now? The bounds are the same like before, since it's calculated already the *new* way.

Reply:
> on the other hand, the bounds are in fact the same. Just the symbol position differs, what is expected.

I'm not 100% confident it's the best we can get (ideally bounds returns the tightest possible rect containing the symbol, or labels will be visually far from their corresponding symbols), but we can always tweak that later if/when needed.

Reply:
Maybe it would be better to have the “real” bounds, yes, but still, this is a fix of a wrong previous behavior, just introduced with a new setting to keep the old (wrong) behavior for backwards compatibility. So I think it's better not to introduce a new behavior here either. But with this PR here #60080 I have started to implement the anchors for the bounds, which would replace the old anchors, except this baseline anchor (but I closed it again, because I continue work on it currently).